assets/src/edit-story/components/form/color.js

```diff
@@ -18,107 +18,184 @@
+  return generatePatternCSS({ color: { r, g, b } });
+}
+
+function getPreviewOpacity(pattern, specifiedOpacity = 1) {
+  if (!pattern) {
+    return specifiedOpacity * 100;
+  }
+  const isSolidPattern = pattern.type === 'solid' || !pattern.type;
+  if (!isSolidPattern) {
+    return specifiedOpacity * 100;
+  }
+  const {
+    color: { a = 1 },
+  } = pattern;
+  return a * 100;
+}
+
+function getPreviewText(pattern) {
+  if (!pattern) {
+    return null;
+  }
+  switch (pattern.type) {
+    case 'radial':
+      return __('Radial', 'web-stories');
+    case 'conic':
+      return __('Conic', 'web-stories');
+    case 'linear':
+      return __('Linear', 'web-stories');
+    case 'solid':
+    default:
+      const {
+        color: { r, g, b, a },
+      } = pattern;
+      if (a === 0) {
+        return null;
+      }
+      return printRGB(r, g, b);
+  }
+}
 
-function ColorInput({
-  onBlur,
-  onChange,
-  isMultiple,
-  opacity,
-  label,
-  value,
-  ...rest
-}) {
-  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
+function ColorInput({ onChange, isMultiple, opacity, value, label }) {
+  const previewStyle = getPreviewStyle(isMultiple ? null : value);
+  const previewText = getPreviewText(value);
+  const previewOpacity = getPreviewOpacity(value, opacity);
+
+  const {
+    actions: { showColorPickerAt, hideColorPicker },
+  } = useColorPicker();
+
+  const ref = useRef();
+
+  const handleOpenEditing = useCallback(() => {
+    showColorPickerAt(ref.current, {
+      color: value,
+      onChange,
+      onClose: hideColorPicker,
+    });
+  }, [showColorPickerAt, hideColorPicker, value, onChange]);
 
   return (
-    <Container>
-      {label}
-      <StyledInput
-        type="color"
-        placeholder={placeholder}
-        label={label}
-        value={value}
-        {...rest}
-        onChange={(evt) => onChange(evt.target.value, evt)}
-        onBlur={(evt) => {
-          if (evt.target.form) {
-            evt.target.form.dispatchEvent(new window.Event('submit'));
-          }
-          if (onBlur) {
-            onBlur();
-          }
-        }}
-      />
-      {value && <span>{value}</span>}
-      {opacity && (
-        <span>
-          {opacity * 100}
-          {'%'}
-        </span>
-      )}
+    <Container ref={ref}>
+      <Preview onClick={handleOpenEditing}>
+        <VisualPreview style={previewStyle} />
+        <TextualPreview aria-label={label}>
+          {isMultiple
+            ? __('Multiple', 'web-stories')
+            : previewText ||
+              _x('None', 'No color or gradient selected', 'web-stories')}
+        </TextualPreview>
+      </Preview>
+      <OpacityPreview isVisible={Boolean(previewText)}>
+        {previewOpacity}
+        {_x('%', 'Percentage', 'web-stories')}
+      </OpacityPreview>
     </Container>
   );
 }
 
 ColorInput.propTypes = {
-  label: PropTypes.string,
-  value: PropTypes.any.isRequired,
+  value: PatternPropType,
```

Comment:
Is `opacity` currently separate from the value? Heads up: I think we need to merge opacity into gradient colors as well as the color multiplier. Layer-level opacity is not the same as color opacity from the structural point of view or browser performance. I'm talking to Sam and Paul about this. I'm ok if this is the first merge, of course - will be just something to redo slightly once that clears up.

Reply:
We can extra common opacity from a gradient and find it that way. But if we're going to have a layer opacity, then that'll fix it - i guess?

Reply:
No. It's currently separate from the layer opacity.  The layer opacity will now go separate, I believe. It's really just an `a` multiplier.

Reply:
Hm, let's merge this (which will be without a specific "global" opacity setting for gradients) and see how it works. If we need an extra opacity slider, we can figure it out later.